rust/src/ikev2/ikev2.rs

```diff
@@ -0,0 +1,715 @@
+                }
+                if hdr.init_spi == 0 {
+                    SCLogInfo!("Malformed SPI - wrong length");
+                    self.set_event(IKEV2Event::MalformedData);
+                    return -1;
+                }
+                // only analyse IKE_SA, other payloads are encrypted
+                if hdr.exch_type != IkeExchangeType::IKE_SA_INIT {
+                    return 0;
+                }
+                let mut tx = self.new_tx();
+                // use init_spi as transaction identifier
+                tx.xid = hdr.init_spi;
+                tx.hdr = (*hdr).clone();
+                match parse_ikev2_payload_list(rem,hdr.next_payload) {
+                    IResult::Done(_,Ok(ref p)) => {
+                        for payload in p {
+                            tx.payload_types.push(payload.hdr.next_payload_type);
+                            match payload.content {
+                                IkeV2PayloadContent::Dummy => (),
+                                IkeV2PayloadContent::SA(ref prop) => {
+                                    // if hdr.flags & IKEV2_FLAG_INITIATOR != 0 {
+                                        self.add_proposals(prop, direction);
+                                    // }
+                                },
+                                IkeV2PayloadContent::KE(ref kex) => {
+                                    SCLogDebug!("KEX {:?}", kex.dh_group);
+                                    if direction == STREAM_TOCLIENT {
+                                        self.dh_group = kex.dh_group;
+                                    }
+                                },
+                                IkeV2PayloadContent::Nonce(ref n) => {
+                                    SCLogDebug!("Nonce: {:?}", n);
+                                },
+                                IkeV2PayloadContent::Notify(ref n) => {
+                                    SCLogDebug!("Notify: {:?}", n);
+                                    if n.notify_type.is_error() {
+                                        tx.errors += 1;
+                                    }
+                                    tx.notify_types.push(n.notify_type);
+                                },
+                                // XXX CertificateRequest
+                                // XXX Certificate
+                                // XXX Authentication
+                                // XXX TSi
+                                // XXX TSr
+                                // XXX IDr
+                                _ => {
+                                    SCLogInfo!("Unknown payload content {:?}", payload.content);
+                                },
+                            }
+                            self.connection_state = self.connection_state.advance(payload);
+                        };
+                    },
+                    e @ _ => SCLogInfo!("parse_ikev2_payload_with_type: {:?}",e),
```

Comment:
what is this notation/syntax?

Reply:
This is a way to name parameters in a match arm (https://doc.rust-lang.org/book/first-edition/patterns.html#bindings).
That said, naming `_` is not useful, this is equivalent to
```rust
match ... {
    e => SCLogInfo!(...)
```
I removed the `@ _` part